kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -186,8 +183,7 @@ private CompletableFuture<ProduceResponse> produce(
               ProduceResponse response =
                   toProduceResponse(
                       clusterId, topicName, keyFormat, keySchema, valueFormat, valueSchema, result);
-              long latency =
-                  Duration.between(requestInstant, result.getCompletionTimestamp()).toMillis();
+              Duration latency = Duration.between(requestStart, result.getCompletionTimestamp());
```

Comment:
The completion timestamp that we create in the callback of `Producer.send(ProducerRecord, Callback)` should also be monotonic.

Reply:
Done.